### PR TITLE
✨ Add CODEOWNERS branch protection check

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Enhance Branch-Protection check to require CODEOWNER review.

This adds 'bonus points' to the Branch-Protection check, but doesn't detract from the score if it's not enabled.

#### What is the current behavior?
Today, the Branch-Protection check requires that code review must be completed before merge. In GitHub, CODEOWNERS can be assigned to subpaths in a. Without CODEOWNER review, anyone with write access to the repo can approve any PR. In large repos with many modules and many maintainers, CODEOWNERS allows maintainers to delegate code reviews and limit the scope of approval. 

#### What is the new behavior (if this is a feature change)?**
Added a unit test; updated existing unit tests.

#### Does this PR introduce a user-facing change?

```release-note
Scorecard will evaluate whether CODEOWNER review is required for protected branches.
```

### CODE REVIEW

File: checks/evaluation/branch_protection.go
Comment (laurentsimon) [inline checks/evaluation/branch_protection.go:33]: this should be a new level if you want a new level. The sum of all the levels should be 10. 

How bout packing in an existing level? 
One issue with adding a level 6 is that it implies a dependence on the previous levels,  meaning it will only be satisfied if all the previous levels are satisfied. 
In practice ,it's hard for us to argue that these levels make sense... we tried to adjust them in a way that sort of does, but it's hard :)

That's why I'm trying to find a place within a previous level to do it. I think we could put it as part of level 4. Level 5 is virtually non-attainable unless you enable force LGTM after each commit in a single PR (which is un-usable)

File: checks/evaluation/branch_protection.go
Comment (laurentsimon) [inline checks/evaluation/branch_protection.go:136]: these function could be re-written with go generic now that they are supported ;)

**Review by laurentsimon — COMMENTED** [review]
LGTm, looks great.
I think we just need to try to put the changes as part of an existing "level", see my comment.

Wdut?

File: checks/evaluation/branch_protection.go
Comment (laurentsimon) [reply checks/evaluation/branch_protection.go]: SGTM.

File: docs/checks/internal/checks.yaml
Comment (olivekl) [inline docs/checks/internal/checks.yaml:201]: Minor nit: can you format this so the list hierarchy is clear? Are these both under `For administrator`? If separate, put `dismiss stale review` under line 201 and indented. If both apply to admins, indent the `require CODEOWNER review` as well.

**Review by olivekl — COMMENTED** [review]
Docs look good to me, with a minor nit for formatting.

**Review by olivekl — APPROVED** [review]

**Comment by naveensrinivasan** [comment]
@raghavkaul  The linter check is failing. Could you please fix that? We should be able to merge after that.

**Comment by naveensrinivasan** [comment]
@raghavkaul  A friendly ping!

File: docs/checks/internal/checks.yaml
Comment (raghavkaul) [reply docs/checks/internal/checks.yaml:206]: Yes! Fixed.

**Pull request merged**
